### PR TITLE
Add scala3 docs translated in Japanese

### PR DESCRIPTION
**Pull request by ItoYo16u**
Hi! I partly translated scala3 docs in Japanese!

If there's anything to change, please tell me.

## TODO
- [x] improve getting-started guide
- [x] translate some sentences yet to be done

### CODE REVIEW

**Comment by ItoYo16u** [comment]
I can not figure out why Ci failed since it works well locally.

**Comment by griggt** [comment]
The CI failure may be related to the recent purge of old documentation: https://github.com/scala/scala-lang/issues/453#issuecomment-776765485

**Comment by ItoYo16u** [comment]
@griggt Thank you! Is there any workaround for this now? or should I just ignore the failure?

File: _ja/index.md [hunk 1]
```diff
@@ -1,13 +1,47 @@
...
+        icon: "fa fa-book"
+        link: /scala3/book/introduction.html
+  - title: "More detailed information"
+    links:
+      - title: "Migration Guide"
+        description: "Scala 2 からScala 3 へ移行するためのガイド"
+        icon: "fa fa-suitcase"
+        link: https://scalacenter.github.io/scala-3-migration-guide/
+      - title: "Guides"
+        description: "Scala3の言語仕様からピックアップして解説"
```
Comment (exoego) [inline _ja/index.md:32]:
```suggestion
        description: "Scala 3 の言語仕様からピックアップして解説"
```

File: _ja/index.md [hunk 2]
```diff
@@ -1,13 +1,47 @@
...
+        icon: "fa fa-rocket"
+        link: /ja/scala3/getting-started.html
+      - title: "Scala 3 Book"
+        description: "主要な言語仕様のイントロダクションをオンラインブックで読む"
+        icon: "fa fa-book"
+        link: /scala3/book/introduction.html
+  - title: "More detailed information"
+    links:
+      - title: "Migration Guide"
+        description: "Scala 2 からScala 3 へ移行するためのガイド"
```
Comment (exoego) [inline _ja/index.md:28]:
```suggestion
        description: "Scala 2 から Scala 3 へ移行するためのガイド"
```

File: _ja/index.md [hunk 3]
```diff
@@ -1,13 +1,47 @@
...
+
+scala3-sections:
+  - title: "First steps"
+    links:
+      - title: "Scala 3 の新機能"
+        description: "Scala 3 で追加されたさまざまな新機能の概要"
+        icon: "fa fa-star"
+        link: /ja/scala3/new-in-scala3.html
+      - title: "Scala 3 をはじめる"
+        description: "あなたのコンピューターにScala 3 をインストールしてScalaコードを書きはじめよう!"
```
Comment (exoego) [inline _ja/index.md:18]:
```suggestion
        description: "あなたのコンピューターに Scala 3 をインストールしてScalaコードを書きはじめよう!"
```

File: _ja/scala3/contribute-to-docs.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+- Macros tutorial
+- Migration guide
+- Scala 3 language reference
+
+ドキュメンテーションの種類に関わらずコミュニティからのコントリビューションを歓迎します。
+
+
+### コントリビューションの仕方
+一般に、 多くの異なった方法で私たちを支援することができます :
+- **Confused about something in any of the docs?** Issue をたててください。
```
Comment (exoego) [inline _ja/scala3/contribute-to-docs.md:23]: 翻訳漏れのようです〜。

```suggestion
- **ドキュメントのどこかで混乱するところがある** Issue をたててください。
```

File: _ja/scala3/contribute-to-docs.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+- **最新の状態を反映していないドキュメントがある** Issueをたてるか、PR をつくってください。
+- **タイポの修正やその他ちょっとした文章の改善** PRをつくってください。
+- **なにかを新しく追加したり大きな変更を加えたい**  議論できるようIssueをたててください。
```
Comment (exoego) [inline _ja/scala3/contribute-to-docs.md:26]:
```suggestion
- **最新の状態を反映していないドキュメントがある** Issue をたてるか、PR をつくってください。
- **タイポの修正やその他ちょっとした文章の改善** PR をつくってください。
- **なにかを新しく追加したり大きな変更を加えたい**  議論できるよう Issue をたててください。
```

半角英単語の前後には半角スペースがあると読みやすいと思います

File: _ja/scala3/contribute-to-docs.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+通常、ドキュメントプロジェクトのそれぞれには編集・改善用のリンクが含まれています。(このドキュメントについても同様で、目次の領域にあります。) また、コントリビューションをはじめるために必要な情報は以下に記載されています。
+
+## Scala 3 Book
+[Scala 3 Book][scala3-book] は Alvin Alexander 氏 が書いています。 この本はScala 3 のすべての重要な機能の概説書です。これから Scala を使いはじめる読者を対象にしています。
```
Comment (exoego) [inline _ja/scala3/contribute-to-docs.md:31]:
```suggestion
[Scala 3 Book][scala3-book] は Alvin Alexander 氏 が書いています。 この本は Scala 3 のすべての重要な機能の概説書です。これから Scala を使いはじめる読者を対象にしています。
```

File: _ja/scala3/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+scala3: true
+language: ja
+---
+
+
+
+## Try Scala without installing anything
+
+いますぐ Scala 3 を試してみたいなら <a href="https://scastie.scala-lang.org/?target=dotty" target="_blank">ブラウザ上で動作する“Scastie” </a> を使ってみてください。
+_Scastie_ はScalaのコードがどのように動作するか試せるオンラインのプレイグラウンドです。全ての Scala のコンパイラと公開されているライブラリを使うこともできます。
```
Comment (exoego) [inline _ja/scala3/getting-started.md:13]:
```suggestion
_Scastie_ は Scala のコードがどのように動作するか試せるオンラインのプレイグラウンドです。全ての Scala のコンパイラと公開されているライブラリを使うこともできます。
```

File: _ja/scala3/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+
+
+## Scala をインストールする
+
+Scala をインストールするということはつまりさまざまなコマンドラインツールやビルドツールをインストールするということです。
+Scalaのインストーラツールの"Coursier"を使うのがおすすめです。Coursierを使えばインストールが必要なものすべてを自動でインストールできます。 もちろん、手動でそれぞれのツールをインストールすることもできます。最低限 Java、sbt があれば Scala のコードを書き始められます。
```
Comment (exoego) [inline _ja/scala3/getting-started.md:19]:
```suggestion
Scala のインストーラツールの "Coursier"を使うのがおすすめです。Coursier を使えばインストールが必要なものすべてを自動でインストールできます。 もちろん、手動でそれぞれのツールをインストールすることもできます。最低限 Java、sbt があれば Scala のコードを書き始められます。
```

File: _ja/scala3/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+
+## Try Scala without installing anything
+
+いますぐ Scala 3 を試してみたいなら <a href="https://scastie.scala-lang.org/?target=dotty" target="_blank">ブラウザ上で動作する“Scastie” </a> を使ってみてください。
+_Scastie_ はScalaのコードがどのように動作するか試せるオンラインのプレイグラウンドです。全ての Scala のコンパイラと公開されているライブラリを使うこともできます。
+
+
+## Scala をインストールする
+
+Scala をインストールするということはつまりさまざまなコマンドラインツールやビルドツールをインストールするということです。
```
Comment (exoego) [inline _ja/scala3/getting-started.md:18]: ひらがなが続いて区切りが分かりにくく思いました
```suggestion
Scala をインストールするということはつまり、さまざまなコマンドラインツールやビルドツールをインストールするということです。
```

File: _ja/scala3/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+次の手順に従って Coursier をインストールしてください。
+
+<div class="main-download">
+  <div id="download-step-one">
+    <p><a href="https://get-coursier.io/docs/cli-overview.html#install-native-launcher" target="_blank"><code>cs</code> launcherをインストールする手順</a>に従ってインストールし、次のコマンドを実行してください。:</p>
+    <p><code>$ ./cs setup</code></p>
+  </div>
+</div>
+
+JVMs の管理に加えて、 `cs setup` コマンドは次のような便利なコマンドラインツールもインストールします。:
```
Comment (exoego) [inline _ja/scala3/getting-started.md:35]: 複数形の `s` が日本語に残っているのは違和感があるので、削除してはどうでしょう

```suggestion
JVM の管理に加えて、 `cs setup` コマンドは次のような便利なコマンドラインツールもインストールします。:
```

File: _ja/scala3/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+1. Java を [Oracle Java 8](https://www.oracle.com/java/technologies/javase-jdk8-downloads.html), [Oracle Java 11](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html), か [AdoptOpenJDK 8/11](https://adoptopenjdk.net/)からダウンロードしてください。 Scala/Java の互換性の詳細については [JDK Compatibility](/overviews/jdk-compatibility/overview.html) を参照してください。
+2.  [sbt](https://www.scala-sbt.org/download.html) をインストールしてください。
+
+## sbtで “Hello, world” プロジェクトを作成する
+
+Scala 3 をインストールする前に、Scala を使ったことがない開発者に向けて sbt を使ってプロジェクトを作成、実行する方法について解説します。
+
+もし sbt でプロジェクトを作成する方法を既に知っているなら 「Scala 3 のインストール」 の章まで読み飛ばしてください。
+
+コマンドライン、IDEのどちらからでも sbt プロジェクトを作成できます。
```
Comment (exoego) [inline _ja/scala3/getting-started.md:68]:
```suggestion
コマンドライン、IDE のどちらからでも sbt プロジェクトを作成できます。
```

File: _ja/scala3/getting-started.md [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+今のところ `src/main/scala` にある `Main.scala` しか必要ありません。
+
+sbt の詳しいドキュメントは [Scala Book](/scala3/book/scala-tools.html) と  sbt 公式 [documentation](https://www.scala-sbt.org/1.x/docs/index.html) にあります。
+
+
+{% comment %}
+### With IntelliJ IDEA
+
+You can skip the rest of this page and go directly to [Building a Scala Project with IntelliJ and sbt](/getting-started/intellij-track/building-a-scala-project-with-intellij-and-sbt.html)
+{% endcomment %}
```
Comment (exoego) [inline _ja/scala3/getting-started.md:107]: 翻訳漏れのようです

File: _ja/scala3/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+```
+
+Scala 2 とクロスコンパイル可能なプロジェクトテンプレートを利用する場合は以下のコマンドを実行してください。
+
+```shell
+sbt new lampepfl/dotty-cross.g8
+```
+#### Coursierを使ってScala 3 用のコンパイラと REPL をインストールする(オプショナル)
+
+以下では Scala 3 のコンパイラ、REPLと Scala 3 をインストールする手順を説明します。
```
Comment (exoego) [inline _ja/scala3/getting-started.md:180]:
```suggestion
以下では Scala 3 のコンパイラ、REPL と Scala 3 をインストールする手順を説明します。
```

File: _ja/scala3/getting-started.md [hunk 14]
```diff
@@ -0,0 +1,250 @@
...
+
+
+
+#### Scala 3 を手動でインストールする(オプショナル)
+
+Scala 3 はまだリリースされていないので Github から最新のソース(2021/02時点で scala3-3.0.0-M3 )を直接ダウンロードしてpathを通してください。
+
+```shell
+wget https://github.com/lampepfl/dotty/releases/download/3.0.0-M3/scala3-3.0.0-M3.tar.gz
+tar -zxvf scala3-3.0.0-M3.tar.gz
```
Comment (exoego) [inline _ja/scala3/getting-started.md:204]: NIT) 最近、RC1 がリリースされました。せっかくなので翻訳版では更新しても良いと思います。

File: _ja/scala3/index.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+        icon: "fa fa-rocket"
+        link: /ja/scala3/getting-started.html
+      - title: "Scala 3 Book"
+        description: "主要な言語仕様のイントロダクションをオンラインブックで読む"
+        icon: "fa fa-book"
+        link: /scala3/book/introduction.html
+  - title: "More detailed information"
+    links:
+      - title: "Migration Guide"
+        description: "Scala 2 からScala 3 へ移行するためのガイド"
```
Comment (exoego) [inline _ja/scala3/index.md:29]:
```suggestion
        description: "Scala 2 から Scala 3 へ移行するためのガイド"
```

File: _ja/scala3/index.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+        icon: "fa fa-suitcase"
+        link: https://scalacenter.github.io/scala-3-migration-guide/
+      - title: "Guides"
+        description: "Scala3の言語仕様からピックアップして解説"
```
Comment (exoego) [inline _ja/scala3/index.md:33]:
```suggestion
        description: "Scala 3 の言語仕様からピックアップして解説"
```

File: _ja/scala3/new-in-scala3.md [hunk 17]
```diff
@@ -0,0 +1,126 @@
+---
+layout: singlepage-overview
+title: New in Scala 3
+scala3: true
+---
+
+Scala 3 はScala 2 から大幅な改善が行われ、さまざまな新機能が追加されています。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:7]:
```suggestion
Scala 3 は Scala 2 から大幅な改善が行われ、さまざまな新機能が追加されています。
```

File: _ja/scala3/new-in-scala3.md [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+ここではScala 3 の特に重要な変更点を概観します。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:8]:
```suggestion
ここでは Scala 3 の特に重要な変更点を概観します。
```

File: _ja/scala3/new-in-scala3.md [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+
+より詳しく知りたい方は以下の参考リンクをご覧ください。
+
+- [Scala 3 Book]({% link _overviews/scala3-book/introduction.md %}) はScalaを書いたことがない開発者向けに書かれています。
+- [Syntax Summary][syntax-summary] ではScala 3 で新しく追加されたシンタックスを解説しています。
+- [Language Reference][reference] を見ればScala 2 と Scala 3の変更点を詳しく確認できます。
+- Scala 2 からScala 3 への移行を考えている方は[Migration Guide][migration] をご覧ください。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:15]:
```suggestion
- [Scala 3 Book]({% link _overviews/scala3-book/introduction.md %}) は Scala を書いたことがない開発者向けに書かれています。
- [Syntax Summary][syntax-summary] では Scala 3 で新しく追加されたシンタックスを解説しています。
- [Language Reference][reference] を見れば Scala 2 と Scala 3 の変更点を詳しく確認できます。
- Scala 2 から Scala 3 への移行を考えている方は[Migration Guide][migration] をご覧ください。
```

File: _ja/scala3/new-in-scala3.md [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+
+## What's new in Scala 3
+Scala 3 は Scala 2 を徹底的に見直して再設計されています。核心部分で、型システムの多くの面が変更されより原理原則に基づいたものになりました。この変更によって新機能(ユニオン型)が使えるようになったこともそうですが、なにより型システムがさらに使いやすくなりました。 例えば、[型推論][type-inference] や overload resolution が改善されました.
+
+### 新機能 & 特長: 文法
+(重要度の低い)多くの文法の整理に加えて、Scala 3 の文法は次のように改善されました。
+
+- 制御構造( `if`, `while`, や `for`)を書く際に括弧を省略してより簡潔に書けるようになりました。 ([new control syntax][syntax-control])
+-  `new` キーワードがオプショナルになりました。 (_aka_ [creator applications][creator])
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:24]:
```suggestion
- `new` キーワードがオプショナルになりました。 (_aka_ [creator applications][creator])
```

File: _ja/scala3/new-in-scala3.md [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+それぞれを組み合わせることで高い(そして時には見たこともないような)表現力を発揮する一連の強力な機能をユーザーにあたえることがScalaの当初のコアコンセプトとしてありました。(これは今でもある程度は当てはまります。) 例えば _implicit_ の機能は、コンテキストの受け渡し、型レベル演算、型クラス、暗黙の変換、既存クラスの拡張メソッドなどに使われています。
+
+これらのユースケースを参考にして、Scala 3 では少し違ったアプローチをとっています。Scala 3 では、`implicit`がどのようなメカニズムによるものか、ということよりもどのような意図で使われるのかに焦点を当てています。
+
+Scala3 ではひとつの強力な機能として`implicit`を提供するのではなく、開発者がその意図を表現しやすいように複数の異なる言語機能として提供しています。
+
+- **Abtracting over contextual information**. [Using clauses][contextual-using]を使って呼び出し時に利用可能で、暗黙に引き渡されるべき情報を抽象化することができます。Scala 2 からの改善としては、`using` 節が型だけで指定できるようになったことが挙げられます。 これによって明示的に参照されることのない関数の引数に命名する必要がなくなりました。
+- **Providing Type-class instances**. [Given instances][contextual-givens] を使ってある型に対応する _canonical value_ を定義することができます。実装を公開することなく、型クラスを使ったプログラミングをよりわかりやすく書けます。。
+
+- **Retroactively extending classes**. In Scala 2 では拡張メソッドは暗黙の変換か implicit classを使って書くことができました. 一方 Scala 3 では [extension methods][contextual-extension] が直接的に言語使用に含まれているのでよりわかりやすいエラーメッセージを表示できます。型推論も改善されました。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:39]:
```suggestion
- **Retroactively extending classes**. Scala 2 では拡張メソッドは暗黙の変換か implicit classを使って書くことができました. 一方 Scala 3 では [extension methods][contextual-extension] が直接的に言語使用に含まれているのでよりわかりやすいエラーメッセージを表示できます。型推論も改善されました。
```

File: _ja/scala3/new-in-scala3.md [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+- **Viewing one type as another**. 暗黙の変換は型クラス`Conversion`のインスタンスとしてゼロから [再設計][contextual-conversions]されました。 
+- **Higher-order contextual abstractions**. 全く新しい 機能である  [context functions][contextual-functions] は暗黙の引数をとる関数型を第一級オブジェクトとして扱います。この機能はライブラリ作者にとって重要です。また、簡潔なドメイン特化言語(DSL)を記述するのにも役立ちます。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:41]:
```suggestion
- **Higher-order contextual abstractions**. 全く新しい機能である  [context functions][contextual-functions] は暗黙の引数をとる関数型を第一級オブジェクトとして扱います。この機能はライブラリ作者にとって重要です。また、簡潔なドメイン特化言語(DSL)を記述するのにも役立ちます。
```

File: _ja/scala3/new-in-scala3.md [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+
+- **Actionable feedback from the compiler**. コンパイラが暗黙の引数の解決に失敗した場合、解決に役立つ [import suggestions](https://www.scala-lang.org/blog/2020/05/05/scala-3-import-suggestions.html) を提示します。
+
+### Say What You Mean: 型システムの改善
+型推論の大幅な改善に加えて、 Scala 3 の型システムは他にも様々な新機能があります。これらの機能は型で不変な値を静的に表現するパワフルな手段を提供します。
+
+- **Enumerations**. [Enums][enums] は case class と上手く組み合わせられるよう、また代数的データ型[algebraic data types][enums-adts]の新しい標準をつくるために再設計されました
+
+- **Opaque Types**.  [opaque type aliases][types-opaque]を使うとパフォーマンス低下の懸念なしに実装の詳細を隠ぺいできます。 Opaque types は値クラスにとってかわる概念です。Opaque types を使うとBoxingのオーバーヘッドを起こすことなく抽象化のバリアを設定できます。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:50]:
```suggestion
- **Opaque Types**.  [opaque type aliases][types-opaque]を使うとパフォーマンス低下の懸念なしに実装の詳細を隠ぺいできます。 Opaque types は値クラスにとってかわる概念です。Opaque types を使うと Boxing のオーバーヘッドを起こすことなく抽象化のバリアを設定できます。
```

File: _ja/scala3/new-in-scala3.md [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+
+- **Intersection and union types**. 型システムの基盤を刷新したことで、新しい型システムの機能が使えるようになりました: 交差型 [intersection types][types-intersection](`A & B` と表記する)のインスタンスは  `A` でありかつ `B`であるような型のインスタンスです。 合併型[union types][types-union](`A | B` と表記する) のインスタンスは `A`または`B`のどちらか一方の型のインスタンスです。 これらの2つの構成体は開発者が継承ヒエラルキー以外の方法で柔軟に型制約を表現できるようにします。
+
+- **Dependent function types**. Scala 2 ではすでに引数の型に応じて返り値の型を変化させることができました。Scala 3 ではこのパターンをさらに抽象化することができ、[dependent function types][types-dependent]を表現することができます。 つまり `type F = (e: Entry) => e.Key` というふうに、返り値の型が引数によって変化するように書けます。
+
+- **Polymorphic function types**. dependent function types のように Scala 2 では型パラメタを受け取るメソッドを定義することができました。 しかし、開発者はこれらのメソッドをさらに抽象化することはできませんでした。Scala 3 では、`[A] => List[A] => List[A]` といった書き方をする[polymorphic function types][types-polymorphic] を使って引数に加えて型引数をとるような関数を抽象化できます。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:56]:
```suggestion
- **Polymorphic function types**. dependent function types のように Scala 2 では型パラーメタを受け取るメソッドを定義することができました。 しかし、開発者はこれらのメソッドをさらに抽象化することはできませんでした。Scala 3 では、`[A] => List[A] => List[A]` といった書き方をする[polymorphic function types][types-polymorphic] を使って引数に加えて型引数をとるような関数を抽象化できます。
```

File: _ja/scala3/new-in-scala3.md [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+
+- **Type lambdas**. Scala 2 で[compiler plugin](https://github.com/typelevel/kind-projector)を使わないと表現できなかった型ラムダは Scala 3 では第一級の機能としてサポートされています。: 型ラムダは補助的な型を定義しないでも高階型引数として引数を受け渡せる型レベルの関数です。
+- **Match types**. 暗黙の型解決を使って型レベル演算をエンコードする代わりに、Scala 3 では型のパターンマッチング[matching on types][types-match]をサポートしています。 型レベルの演算を型チェックと統合することでエラーメッセージをわかりやすく改善し、また複雑なエンコーディングをしなくていいようにしています。
+
+
+### 再構想: オブジェクト指向プログラミング
+Scala は常に関数型プログラミングとオブジェクト指向プログラミングの間のフロンティアにあります。そしてScala 3 はその境界を両方に広げます。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:63]:
```suggestion
Scala は常に関数型プログラミングとオブジェクト指向プログラミングの間のフロンティアにあります。そして Scala 3 はその境界を両方に広げます。
```

File: _ja/scala3/new-in-scala3.md [hunk 26]
```diff
@@ -0,0 +1,126 @@
...
+先に言及した型システムの変更とcontextual abtstractions の再設計によって、関数型プログラミングを以前にも増して簡単に書けるようになりました。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:64]:
```suggestion
先に言及した型システムの変更 とcontextual abtstractions の再設計によって、関数型プログラミングを以前にも増して簡単に書けるようになりました。
```

File: _ja/scala3/new-in-scala3.md [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+同時に、次に掲げる新機能を使うと _オブジェクト指向設計_ をうまく構造化してベストプラクティスを実践しやすくなります。
+- **Pass it on**. Traits は class のように 引数をとれるようになりました。詳しくは [parameters][oo-trait-parameters] をご覧ください。 これによってtraitはソフトウェアをモジュールに分解するツールとしてよりいっそうパワフルになりました。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:66]: 日本語では複数形の `s` はないほうが良いと思います（あると別の単語のように一瞬思ってしまいました）

```suggestion
- **Pass it on**. Trait は class のように 引数をとれるようになりました。詳しくは [parameters][oo-trait-parameters] をご覧ください。 これによって trait はソフトウェアをモジュールに分解するツールとしてよりいっそうパワフルになりました。
```

File: _ja/scala3/new-in-scala3.md [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+- **Plan for extension**.  継承を意図していないクラスが継承されてしまうことはオブジェクト指向設計において長年の問題でした。この問題に対処するため Scala 3 では [open classes][oo-open]の概念を導入することで _明示的に_ クラスを継承可能であるとしめすようライブラリ作者に要求するようにしました。
+- **Hide implementation details**. ふるまいを実装した Utility traits は推論される型に含まれるべきでない場合があります。Scala 3 ではそのようなtraitsに [transparent][oo-transparent] とマークすることで継承をユーザーに公開しないようにすることができます。
+- **Composition over inheritance**. このフレーズはしばしば引用されますが、実装するのは面倒です。 しかしScala 3 の [export clauses][oo-export]を使えば楽になります。imports と対称的に、 export clauses はオブジェクトの特定のメンバーへアクセスするためのエイリアスを定義します。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:69]:
```suggestion
- **Composition over inheritance**. このフレーズはしばしば引用されますが、実装するのは面倒です。 しかし　Scala 3 の [export clauses][oo-export]を使えば楽になります。imports と対称的に、 export clauses はオブジェクトの特定のメンバーへアクセスするためのエイリアスを定義します。
```

File: _ja/scala3/new-in-scala3.md [hunk 29]
```diff
@@ -0,0 +1,126 @@
...
+- **No more NPEs**. Scala 3 はかつてないほどnull安全です。: [explicit null][oo-explicit-null] によって `null` を型ヒエラルキーの外側に追い出しました。これによってエラーを静的にキャッチしやすくなります。また、 [safe initialization][oo-safe-init]の追加的なチェックで初期化されていないオブジェクトへのアクセスを検知できます。
```
Comment (exoego) [inline _ja/scala3/new-in-scala3.md:70]:
```suggestion
- **No more NPEs**. Scala 3 はかつてないほど null 安全です。: [explicit null][oo-explicit-null] によって `null` を型ヒエラルキーの外側に追い出しました。これによってエラーを静的にキャッチしやすくなります。また、 [safe initialization][oo-safe-init]の追加的なチェックで初期化されていないオブジェクトへのアクセスを検知できます。
```

Comment (ItoYo16u) [reply _ja/scala3/new-in-scala3.md]: Thank you for your corrections!

File: _ja/scala3/new-in-scala3.md (hunk 24 above)
Comment (ItoYo16u) [reply _ja/scala3/new-in-scala3.md:56]: 「パラメタ」または「パラメータ」が一般的だと思うのですがどうでしょうか。

File: _ja/scala3/getting-started.md (hunk 14 above)
Comment (ItoYo16u) [reply _ja/scala3/getting-started.md]: RC1を反映しました

File: _ja/scala3/new-in-scala3.md (hunk 24 above)
Comment (exoego) [reply _ja/scala3/new-in-scala3.md:56]: あ、本当ですね、パラメータにしたつもりが TYPO してました…
```suggestion
- **Polymorphic function types**. dependent function types のように Scala 2 では型パラメータを受け取るメソッドを定義することができました。 しかし、開発者はこれらのメソッドをさらに抽象化することはできませんでした。Scala 3 では、`[A] => List[A] => List[A]` といった書き方をする[polymorphic function types][types-polymorphic] を使って引数に加えて型引数をとるような関数を抽象化できます。
```

[日本語訳では "パラメータ" の訳が使われており、"パラメタ" は見当たらない](https://github.com/scala/docs.scala-lang/search?q=%E3%83%91%E3%83%A9%E3%83%A1%E3%83%BC%E3%82%BF)ようなので、パラメータにするのがよさそうです。

**Review by exoego — APPROVED** [review]
LGTM from Japanese speaker POV

**Comment by SethTisue** [comment]
@eed3si9n want to give it a once-over, or shall we just merge it at this point?

**Comment by eed3si9n** [comment]
@SethTisue I might be able to nipick a bunch of things, but I'm in favor of merging since it went through a good review by exoego-san already.

Something to note is that the tone is adopted by @ItoYo16u -san is a polite form. In Japanese there are different degrees of politeness built into the basic grammar, and when I lead the translation of Scala 2.x documentation (for instance at https://github.com/scala/docs.scala-lang/blob/master/_ja/overviews/collections/overview.md) I intentionally chose plain form (常体) often called da-dearu style (だ・である調). By convention, academic papers and technical reports are written using the plain form, including the Japanese translation for Odersky, Spoon, Venners 'Programming in Scala.'

On the other hand, the translation here uses polite form (敬体) or desu-masu style (です・ます調). The polite form is the business casual of the Japanese language, and blogs including corporate blogs like [Google Japan](https://japan.googleblog.com/2021/03/IWD-impact-challenge.html) and emails often use it. Ruby's entire [reference manual](https://docs.ruby-lang.org/ja/2.7.0/doc/spec=2fintro.html) is in polite form as well. It's the safe default, but in my opinion it sounds more like a corporate blog post and less like a technical document.

Again, I'm not vetoing this PR because of this, but I feel like this is something we should be aware of, and potentially revisit later.

**Comment by ItoYo16u** [comment]
Thank you for your advice. I will rewrite in da-dearu style soon.

**Comment by ItoYo16u** [comment]
I finished rewriting in plain form.

**Review by eed3si9n — APPROVED** [review]
@ItoYo16u コントリビューションありがとうございます。Thanks for the contribution! 🙇

**Pull request merged**